File: src/dataset/javascript/triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_triangle_v1.js
/* Classifies triangles */
var Triangle = function(a, b, c) {
    if (a <= 0 || b <= 0 || c <= 0) {
        throw "Only positive length sides are allowed";
    }

    if (a + b <= c || a + c <= b || b + c <= a) {
        throw "Not a valid triangle";
    }

    this.triangle = [a, b, c].sort((x, y) => x - y);
};

Triangle.prototype.kind = function() {
    var a = this.triangle[0];
    var b = this.triangle[1];
    var c = this.triangle[2];

    if (a === c) {
        return "equilateral";
    }

    if (a === b || b === c) {
        return "isosceles";
    }

    return "scalene";
};

module.exports = Triangle;

File: src/dataset/python/triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_triangle_v1.py
"""Validate and specify triangle types"""

class TriangleError(Exception):
    """Error if it cannot be a triangle"""
    def __init__(self, message):
        super().__init__(message)

class Triangle(object):
    """Validate and specify triangle types"""
    def __init__(self, a, b, c):
        """Create and validate triangle"""
        sides = sorted((a, b, c))
        if sides[0] + sides[1] <= sides[2]:
            raise TriangleError("Not a Triangle")
        self.a = sides[0]
        self.b = sides[1]
        self.c = sides[2]

    def kind(self):
        """Specify triangle type"""
        if self.a == self.c:
            return "equilateral"
        elif self.a == self.b or self.b == self.c:
            return "isosceles"
        else:
            return "scalene"